src/videojs-contrib-hls.js

```diff
@@ -604,11 +620,11 @@ Hls.comparePlaylistResolution = function(left, right) {
 };
 
 HlsSourceHandler.canPlayType = function(type) {
-  let mpegurlRE = /^application\/(?:x-|vnd\.apple\.)mpegurl/i;
+  let mpegurlRE = /^(audio|video|application)\/(x-|vnd\.apple\.)?mpegurl/i;
 
   // favor native HLS support if it's available
   if (Hls.supportsNativeHls) {
-    return false;
+    return '';
   }
   return mpegurlRE.test(type);
```

Comment:
Maybe to prevent confusion we should still return false in this function, since empty string is going to be used as false in this case, and since we return a boolean below anyway.